product_code/matriculation_service/impl/src/main/scala/de/upb/cs/uc4/matriculation/impl/MatriculationServiceImpl.scala

```diff
@@ -172,6 +200,55 @@ class MatriculationServiceImpl(
         }
     }
 
+  /** Returns a pdf with the certificate of enrollment */
+  override def getCertificateOfEnrollment(username: String, semesterBase64: Option[String]): ServiceCall[NotUsed, ByteString] =
+    identifiedAuthenticated(AuthenticationRole.Student) {
+      (authUsername, _) =>
+        ServerServiceCall { (header, _) =>
+          if (authUsername != username) {
+            throw UC4Exception.OwnerMismatch
+          }
+          var pdf = certificateEnrollmentHtml
+
+          userService.getUser(username).handleRequestHeader(addAuthenticationHeader(header)).invoke().flatMap { user =>
+            getMatriculationData(username).handleRequestHeader(addAuthenticationHeader(header)).invoke().flatMap { data =>
+
+              val semester = if (semesterBase64.isDefined) {
+                new String(Base64.getUrlDecoder.decode(semesterBase64.get), StandardCharsets.UTF_8)
+              }
```

Comment:
If we get a semester here, for which there is no enrollment, we will generate a certificate without fields of study. I think this should be changed.

Reply:
If we cannot find an enrollment we will now throw a `NotFound` exception